src/client/core/utils/array.js

```diff
@@ -0,0 +1,71 @@
+export function toArray (obj) {
+    return Array.prototype.slice.call(obj);
+}
+
+export function isArray (obj) {
+    return obj instanceof Array;
+}
+
+export function indexOf (arr, el) {
+    var index  = -1;
+    var length = arr.length;
+
+    if (!length)
+        return index;
+
+    while (index < length) {
+        if (arr[index] === el)
+            return index;
+
+        index++;
+    }
+
+    return -1;
+}
+
+export function forEach (arr, iterator) {
+    var length = arr.length;
+    var stop   = false;
+
+    for (var i = 0; i < length; i++) {
+        stop = iterator(arr[i], i, arr);
+
+        if (stop === false)
```

Comment:
Don't add additional semantics like `stop`, it will render it unusable for the arrows, e.g.:
```js
forEach(array, item => /* something that returns boolean*/);
```
Use regular cycles if you need conditional stop.